api/v1alpha1/dependencyupdatecheck_types.go

```diff
@@ -28,8 +28,10 @@ type DependencyUpdateCheckSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of DependencyUpdateCheck. Edit dependencyupdatecheck_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// Specifies which components to run Renovate on. The expected format is "workspace/component".
+	// If omitted, Renovate will be run on all components
+	// +optional
+	Components []string `json:"components,omitempty"`
```

Comment:
I'm thinking about how to use a more flexible field for specifying components, and provide the functionality for validating the CR while creating it. What do you think about defining a schema in format like this:
```
workspaces:
  workspace1:
    applications:
      app1:
        components:
        - component1
        - component2
  workspace2:
    applications:
      app1:
        components:
        - componentA
        - componentB
      app2:
        components:
        - component1
        - component2
```
Note: I believe it's possible to create components with same name in different applications, and application with same name in different workspaces.

Reply:
I changed the input to a nested structure and added some kubebuilder validations